monkey/infection_monkey/island_api_client/http_island_api_client.py

```diff
@@ -27,7 +35,15 @@ def handle_island_errors(fn):
     def decorated(*args, **kwargs):
         try:
             return fn(*args, **kwargs)
-        except requests.exceptions.ConnectionError as err:
+        except (
+            IslandAPIConnectionError,
+            IslandAPIError,
+            IslandAPIRequestError,
+            IslandAPIRequestFailedError,
+            IslandAPITimeoutError,
+        ) as e:
+            raise e
+        except (requests.exceptions.ConnectionError, requests.exceptions.TooManyRedirects) as err:
```

Comment:
These all inherit from `IslandAPIError`, so you can just 

```suggestion
        except IslandAPIError as err:
            raise err
        except (requests.exceptions.ConnectionError, requests.exceptions.TooManyRedirects) as err:
```

Also, use `err` not `e` for consistency. It's written somewhere in our guidelines.

Reply:
Nice!